### ISSUE TITLE
'Publishing packages' – Publishing prereleases – should we demonstrate w/ -dev?

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/site-www#2453]
Page URL: https://dart.dev/tools/pub/publishing#publishing-prereleases
Page source: https://github.com/dart-lang/site-www/tree/master/src/tools/pub/publishing.md

We tend to use `-dev` as a suffix for packages during development – only changing it before we publish. This makes it easy for someone looking at the source to understand the state of the package. See https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance

First, should be generalize this convention?

If yes, we should update this page. Use `-alpha` or `-beta`

If yes, we may also want to go as far as adding a warning (or error) to the pub client.

CC @mit-mit @jakemac53 @natebosch @kwalrath

**Comment by mit-mit** [comment]
I'm confused, because we have an internal team practice on a few packages, we now want to change our external guidance for how our customers should do it?

I like the current guidance; using `-dev` is aligned with the version numbers we use for the core SDK: https://dart.dev/get-dart#release-channels

Wouldn't it be simpler to change our internal practice to use something like `-unpublished`?

**Comment by jakemac53** [comment]
I don't think we need to be overly prescriptive here, I would definitely not add a warning or error to pub.

Given that we are going to start publishing pre-release versions then I agree with @mit-mit that we should reconsider how we are managing the "in progress" versions as the current `-dev` strategy would become confusing I think.

**Comment by kevmoo** [comment]
I'm fine going either way – just pushing for consistency here.

A lot of folks follow along w/ our repos and assume we're always using best practices.

**Comment by kevmoo** [comment]
Didn't mean to close!

**Comment by atsansone** [comment]
@mit-mit , @kevmoo , @jakemac53 : Is this still a going concern? If so, do you have any updates?

**Comment by mit-mit** [comment]
`-wip` sgmt. one other idea is `-next`

**Comment by kevmoo** [comment]
I like `-wip`

**Comment by jakemac53** [comment]
No opinion here, happy to use whatever suffix.

**Comment by parlough** [comment]
Sounds like there has been consensus on using `-wip` and a bunch of Dart team projects have migrated to doing so.

Unless someone chimes in, the task here is now:

- Update docs to suggest `-wip` as suffix for in-progress work and `-dev` for prereleases
